guides/plugins/plugins/framework/event/finding-events.md

```diff
@@ -0,0 +1,184 @@
+This way you can also find out about all the possible DAL events available in Shopware.
+
+Finding those "event classes" can be done by searching for the term `@Event` in your project.
+
+You can use those events in a [subscriber](../../plugin-fundamentals/listening-to-events.md) like the following:
+```php
+public static function getSubscribedEvents(): array
+{
+    return [
+        ProductEvents::PRODUCT_LOADED_EVENT => 'onProductsLoaded',
+        'custom_entity.written' => 'onCustomEntityWritten'
+    ];
+}
+```
+
+As you can see, you can either use the event class constants, if available, or the string itself.
+
+You'll then have access to several event specific information, e.g. your listener method will have access to an [EntityWrittenEvent](https://github.com/shopware/platform/blob/v6.4.0.0/src/Core/Framework/DataAbstractionLayer/Event/EntityWrittenEvent.php)
+instance when subscribing to the `written` event.
+
+```php
+public function onCustomEntityWritten(EntityWrittenEvent $event): void
+{
+}
+```
+
+You can find all of those DAL event classes [here](https://github.com/shopware/platform/tree/v6.4.0.0/src/Core/Framework/DataAbstractionLayer/Event).
+
+## Other PHP events
+
+If the [DAL events](#DAL events) do not suffice for your use case, there are a few other events built into Shopware.
+These are not auto-generated events, but rather events we built in with purpose.
+
+There are multiple ways to find them:
+- By actually looking at the code, that you want to extend
+- By specifically searching for them
+- By having a look at the service definition of a given class
+
+### Looking at the code
+
+You will most likely look into our Core code quite a lot, while trying to understand what's happening and why things are happening.
+On your journey looking through the code, you may stumple upon code looking like this:
+
+```php
+$someEvent = new SomeEvent($parameters, $moreParameters);
+$this->eventDispatcher->dispatch($someEvent, $someEvent->getName());
+```
+
+This is an event that's being fired manually, which you can react upon.
+Make sure to always have a look at the event class itself in order to find out which information it contains.
+
+When subscribing to those events, your event listener method will have access to the previously created event instance. 
+
+```php
+public static function getSubscribedEvents(): array
+{
+    return [
+        SomeEvent::class => 'registeringToSomeEvent'
+    ];
+}
+
+public function registeringToSomeEvent(SomeEvent $event): void
+{
+}
+```
+
+The [next section](#Specifically searching for events) will cover how to find those events without randomly stumbling upon them.
+
+### Specifically searching for events
+
+If you're really looking for a fitting event for your purpose, you might want to directly search for them.
+This can be done by searching through the `<shopware root>/platform/src` or the `<shopware root>/vendor/shopware/platform/src` directory,
```

Comment:
That is difficult. When you are working with Shopware as a dependency (what we do with the production template or our HEPTAconnect SDK) we do not require shopware/platform but their clones shopware/core and shopware/storefront and therefore have no /src directory in there.

Reply:
Hm, any ideas how to "fix" that?
Maybe smth like `<core root>/...` or `<administration root>/...`?

Reply:
How is it done elsewhere? I mean this is probably not the first text that references these places. In your situation I would rely on development and production template as these are the ways you support out of the box. Therefore use platform reference for development and shopware/core for production references.